yt_idv/shaders/ray_tracing.frag.glsl

```diff
@@ -53,6 +158,22 @@ void spherical_coord_shortcircuit()
     vec3 dir = -normalize(camera_pos.xyz - ray_position_xyz);
     vec4 curr_color = vec4(0.0);
 
+    // intersections
+    vec2 t_sphere_outer = get_ray_sphere_intersection(right_edge[id_r], ray_position_xyz, dir);
+    if (t_sphere_outer[0] == INFINITY && t_sphere_outer[1] == INFINITY)
+    {
+        // if there are no intersections with the outer sphere, then there
+        // will be no intersections with the remaining faces and we can stop
+        // looking.
+        discard;
+    }
+
+    vec2 t_sphere_inner = get_ray_sphere_intersection(left_edge[id_r], ray_position_xyz, dir);
+    float t_p_1 = get_ray_plane_intersection(vec3(phi_plane_le), phi_plane_le[3], ray_position_xyz, dir);
+    float t_p_2 = get_ray_plane_intersection(vec3(phi_plane_re), phi_plane_re[3], ray_position_xyz, dir);
+    vec2 t_cone_outer = get_ray_cone_intersection(right_edge[id_theta], ray_position_xyz, dir);
+    vec2 t_cone_inner= get_ray_cone_intersection(left_edge[id_theta], ray_position_xyz, dir);
+
```

Comment:
so once we have these, what do we do?

Reply:
1. all those t values are the intersections with the base geometries on which the faces of the spherical voxel lies, so we need to find the t values that actually intersect the voxel, not just the base geometries. Those would be the ray entry/exit points of the current voxel. 
2. I think those entry/exit points correspond to t_min and t_max of the standard cartesian tracing? So we'd similarly want to take n_samples of the current fragment between those min/max values?